<img src="tpan1.png">

### Résumé
>Dans ces MINI-Projet nous avons accédé aux données de référence sur les fréquences en France. Cette donnée renseigne l'emplacement des antennes relais ainsi que les mesures des champs électromagnétiques sur le département de l'Hérault, pour l'analyser est interprété des résultats et nous avons fait des prédictions en utilisons des algorithmes machinés learning. 	&#x1F601;



* **<span style="color:red">Ces données sont la propriété de l'Agence Nationale des Fréquences (ANFR) </span>**

In [2]:
import pandas as pd 
import numpy as np
import csv
import matplotlib.pyplot as plt
import seaborn as sns
from csv import writer
import random
df=pd.read_csv('ANFR_Herault_Mesures_originales_pour_jpynb.csv', delimiter=';')
df.head()

Numero de mesure  Longitude  Latitude  \
0               175    3.91611  43.58528   
1               180    4.04194  43.64694   
2               543    3.03167  43.36028   
3               545    3.03278  43.36167   
4               850    3.56444  43.64583   

                                    Position  Insee  \
0  03e 54' 58.00'' Est, 43e 35' 07.00'' Nord  34129   
1  04e 02' 31.00'' Est, 43e 38' 49.00'' Nord  34176   
2  03e 01' 54.00'' Est, 43e 21' 37.00'' Nord  34225   
3  03e 01' 58.00'' Est, 43e 21' 42.00'' Nord  34225   
4  03e 33' 52.00'' Est, 43e 38' 45.00'' Nord  34114   

                               Adresse                 Complement  \
0                           Mas de Gau                        NaN   
1  5539 Rue Lou Claou de la Oulivettas                        NaN   
2                 Chemin de Montfaucon  residence de  Mme SOUTADE   
3                   Chemin de Quarante  residence de  M. GONZALEZ   
4           480 Chemin de Pioch Courbi      Chez Mme et M. GALOIS   

   Code postal       Commune Date de mesure Mesure realisee  \
0        34970        LATTES     06/12/2001    En exterieur   
1        34130      MUDAISON     06/12/2001    En exterieur   
2        34620  PUISSERGUIER     23/05/2002    En exterieur   
3        34620  PUISSERGUIER     23/05/2002    En exterieur   
4        34150        GIGNAC     10/10/2002    En exterieur   

              Rapport Conformite de la mesure  Niveau global  \
0  FicheMesure175.pdf                     Oui           0.00   
1  FicheMesure180.pdf                     Oui           0.00   
2  FicheMesure543.pdf                     Oui           0.54   
3  FicheMesure545.pdf                     Oui           0.20   
4  FicheMesure850.pdf                     Oui           0.53   

    Type d'environnement Laboratoire  
0                 Divers     SATELIT  
1                 Divers     SATELIT  
2                 Divers       APAVE  
3                 Divers       APAVE  
4  Rue / Route / Parking       APAVE


 **<span style="color:#0000C0"> →Nous avons bien importé les donnés donc maitenent on peut manipulé les donnés &#x1F600;</span>**
* <p>Comme premier etap j'ai changé le séparateur de mon fichier Csv de `;` vers `,` Afin de ne pas avoir de difficultés dons l'etape suivant

In [3]:
with  open(r"ANFR_Herault_Mesures_originales_pour_jpynb_C.csv") as in_file,  open(r"ANFR_Herault_Mesures_originales_pour_jpynb_V.csv", 'w') as out_file:
    semicolonin = csv.reader(in_file, delimiter=';')
    commaout = csv.writer(out_file, delimiter=',')
    for row in semicolonin:
        commaout.writerow(row)

In [4]:
dframe=pd.read_csv('ANFR_Herault_Mesures_originales_pour_jpynb_V1.csv')
dframe.head()

Numero de mesure  Longitude  Latitude  \
0               175    3.91611  43.58528   
1               180    4.04194  43.64694   
2               543    3.03167  43.36028   
3               545    3.03278  43.36167   
4               850    3.56444  43.64583   

                                    Position  Insee  \
0  03e 54' 58.00'' Est, 43e 35' 07.00'' Nord  34129   
1  04e 02' 31.00'' Est, 43e 38' 49.00'' Nord  34176   
2  03e 01' 54.00'' Est, 43e 21' 37.00'' Nord  34225   
3  03e 01' 58.00'' Est, 43e 21' 42.00'' Nord  34225   
4  03e 33' 52.00'' Est, 43e 38' 45.00'' Nord  34114   

                               Adresse                 Complement  \
0                           Mas de Gau                        NaN   
1  5539 Rue Lou Claou de la Oulivettas                        NaN   
2                 Chemin de Montfaucon  residence de  Mme SOUTADE   
3                   Chemin de Quarante  residence de  M. GONZALEZ   
4           480 Chemin de Pioch Courbi      Chez Mme et M. GALOIS   

   Code postal       Commune Date de mesure Mesure realisee  \
0        34970        LATTES     06/12/2001    En exterieur   
1        34130      MUDAISON     06/12/2001    En exterieur   
2        34620  PUISSERGUIER     23/05/2002    En exterieur   
3        34620  PUISSERGUIER     23/05/2002    En exterieur   
4        34150        GIGNAC     10/10/2002    En exterieur   

              Rapport Conformite de la mesure  Niveau global  \
0  FicheMesure175.pdf                     Oui           0.00   
1  FicheMesure180.pdf                     Oui           0.00   
2  FicheMesure543.pdf                     Oui           0.54   
3  FicheMesure545.pdf                     Oui           0.20   
4  FicheMesure850.pdf                     Oui           0.53   

    Type d'environnement Laboratoire  
0                 Divers     SATELIT  
1                 Divers     SATELIT  
2                 Divers       APAVE  
3                 Divers       APAVE  
4  Rue / Route / Parking       APAVE

In [137]:
dframe.rename(columns={'Niveau global': 'Niveau_global'}, inplace=True)

In [5]:
dframe=dframe.assign(target=1)

In [6]:
dframe.dtypes

Numero de mesure             int64
Longitude                  float64
Latitude                   float64
Position                    object
Insee                        int64
Adresse                     object
Complement                  object
Code postal                  int64
Commune                     object
Date de mesure              object
Mesure realisee             object
Rapport                     object
Conformite de la mesure     object
Niveau global              float64
Type d'environnement        object
Laboratoire                 object
target                       int64
dtype: object

In [7]:
dframe.count()

Numero de mesure           4922
Longitude                  4922
Latitude                   4922
Position                   4922
Insee                      4922
Adresse                    4922
Complement                 4404
Code postal                4922
Commune                    4922
Date de mesure             4922
Mesure realisee            4922
Rapport                    4922
Conformite de la mesure    4922
Niveau global              4922
Type d'environnement       4922
Laboratoire                4922
target                     4922
dtype: int64

<center>---------------------------------</center>
<br><strong>→Dans ces étapes je veux ajouter des nouvelles données au fichier Csv et puisque nous concentrés a la colonne niveau global et Numéro de mesure j'ai les donne des valeurs aléatoires:</strong>
<ol> 
    <li>J'ai donné des fréquences aléatoire réel entre 1 et 7 pour que les valeurs inférieur à 6V/m devient plus probable que les valeur supérieur a 6V/m.</li>
    <li>pour la colonne niveau global l'aure des valeurs aléatoire entier entre 1000 et 9000.</li>
</ol> 

In [8]:
def append_list_as_row(file_name, list_of_elem):
    # Ouvrir le fichier en mode ajout
    with open(file_name, 'a+', newline='') as write_obj:
        # Créer un objet écrivain à partir du module csv
        csv_writer = writer(write_obj)
        # Ajouter le contenu de la liste à la dernière ligne du fichier csv
        csv_writer.writerow(list_of_elem)

In [290]:
def add_line():
    for i in range(0,1000):
        Niveau_glb = random.uniform(1,7)
        Numéro_de_mesure =random.randint(1000,9000)
        row_contents = [Numéro_de_mesure, 3.91611,43.64694,'Est 03e 28 49.64 50.65 Nord',34176,
       'Chemin de Montfaucon,','residence de Mme SOUTADE',34130,'LATTES','06/12/2001','En exterieure', 
                    'FicheMesure175.pdf','Oui', Niveau_glb, 'Divers', 'SATELIT']
        # Ajout d'une ligne à csv avec des entrées manquantes
        append_list_as_row('ANFR_Herault_Mesures_originales_pour_jpynb_V1.csv', row_contents)

In [291]:
add_line()

In [9]:
df2=pd.read_csv('ANFR_Herault_Mesures_originales_pour_jpynb_V1.csv')
df2.tail()

Numero de mesure  Longitude  Latitude                     Position  \
4917              3474    3.91611  43.64694  Est 03e 28 49.64 50.65 Nord   
4918              3921    3.91611  43.64694  Est 03e 28 49.64 50.65 Nord   
4919              3586    3.91611  43.64694  Est 03e 28 49.64 50.65 Nord   
4920              2956    3.91611  43.64694  Est 03e 28 49.64 50.65 Nord   
4921              2728    3.91611  43.64694  Est 03e 28 49.64 50.65 Nord   

      Insee                Adresse                Complement  Code postal  \
4917  34176  Chemin de Montfaucon,  residence de Mme SOUTADE        34130   
4918  34176  Chemin de Montfaucon,  residence de Mme SOUTADE        34130   
4919  34176  Chemin de Montfaucon,  residence de Mme SOUTADE        34130   
4920  34176  Chemin de Montfaucon,  residence de Mme SOUTADE        34130   
4921  34176  Chemin de Montfaucon,  residence de Mme SOUTADE        34130   

     Commune Date de mesure Mesure realisee             Rapport  \
4917  LATTES     06/12/2001   En exterieure  FicheMesure175.pdf   
4918  LATTES     06/12/2001   En exterieure  FicheMesure175.pdf   
4919  LATTES     06/12/2001   En exterieure  FicheMesure175.pdf   
4920  LATTES     06/12/2001   En exterieure  FicheMesure175.pdf   
4921  LATTES     06/12/2001   En exterieure  FicheMesure175.pdf   

     Conformite de la mesure  Niveau global Type d'environnement Laboratoire  
4917                     Oui       2.841814               Divers     SATELIT  
4918                     Oui       1.999765               Divers     SATELIT  
4919                     Oui       4.746528               Divers     SATELIT  
4920                     Oui       4.094137               Divers     SATELIT  
4921                     Oui       1.708546               Divers     SATELIT

In [81]:
df2.count()

Numero de mesure           4922
Longitude                  4922
Latitude                   4922
Position                   4922
Insee                      4922
Adresse                    4922
Complement                 4404
Code postal                4922
Commune                    4922
Date de mesure             4922
Mesure realisee            4922
Rapport                    4922
Conformite de la mesure    4922
Niveau global              4922
Type d'environnement       4922
Laboratoire                4922
dtype: int64

<strong>→Des nouvelles données ont été ajoutées&#x1F601; .</strong>
<br><center>----------------------------------------------</center>

In [10]:
df2.rename(columns={'Niveau global': 'Niveau_global'}, inplace=True)

 **<span style="color:#0000C0"> →Je veux créer une nouvelle colonne "target" en l'affecter à 1.</span>**

 <div class="alert alert-block alert-danger">** <span style="color:#800000">→ Lorsque le niveau d’exposition évalué selon le `cas A` du protocole dépasse le niveau d’attention de <strong>6 V/m</strong>, une évaluation selon le `cas B` du protocole devient obligatoire.  </span>** </div>




In [11]:
df2=df2.assign(target=1)

>Cette fonction nous permet de filtrer où le niveau d'attention est supérieur à 6V/m.

* le labelle '0' si le mesure est supérieur à 6V/m
* resté '1' si le mesure est inférieur à 6V/m

In [12]:
def target():
    m=[]
    for t in range(0,len(df)):
        if df2['Niveau_global'][t] > 6 :
            df2['target'][t] = 0
            m.append(df2['Niveau_global'][t])
    return(m)

In [13]:
targ=target()
print(targ)

[9.63, 8.44, 8.03, 9.06, 9.05, 6.65, 15.66, 25.39, 10.33, 9.1, 6.98, 21.03, 10.57, 8.52]


C:\Users\ASUS\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


In [14]:
pd.crosstab(df2.target, "freq")

col_0   freq
target      
0         14
1       4908

<strong>→d'après le résultat du tableau au-dessus j'ai remarqué que le data set ils ne sont pas équilibrés.</strong> 
> car il y a que 14 valeur supérieur à 6V/m est 4908 inférieur a 6V/m  &#x1F60F;

In [15]:
X=df2[['Numero de mesure','Niveau_global']].values
print(X)
y=df2['target'].to_numpy()
print(y)

[[1.75000000e+02 0.00000000e+00]
 [1.80000000e+02 0.00000000e+00]
 [5.43000000e+02 5.40000000e-01]
 ...
 [3.58600000e+03 4.74652768e+00]
 [2.95600000e+03 4.09413697e+00]
 [2.72800000e+03 1.70854568e+00]]
[1 1 1 ... 1 1 1]


In [16]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)

In [17]:
from sklearn.linear_model import LogisticRegression 
#Create a l.Reg Classifier
logreg1 = LogisticRegression()
#Train the model using the training sets
logreg1.fit(X_train, y_train)
y_pred=logreg1.predict(X_test)

C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


In [18]:
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[   1    4]
 [   0 1472]]
              precision    recall  f1-score   support

           0       1.00      0.20      0.33         5
           1       1.00      1.00      1.00      1472

   micro avg       1.00      1.00      1.00      1477
   macro avg       1.00      0.60      0.67      1477
weighted avg       1.00      1.00      1.00      1477

0.997291807718348


* d'aprés le matrice de confusion ona :
     * ligne 1 colonne 1 : les 0 positive c-à-d les cas où la prédiction est 0, et où la valeur réelle est effectivement 0.
     * ligne 2 colonne 2 : les 1 positive c-à-d les cas où la prédiction est 1, et où la valeur réelle est effectivement 1.
     * ligne 1 colonne 2 : les 1 négative c-à-d les cas où la prédiction est 1, et où la valeur réelle est effectivement 0.
     * ligne 2 colonne 1 : les 0 négative c-à-d les cas où la prédiction est 0, et où la valeur réelle est effectivement 1.
> <br>**<span style="color:#0000C0">Conclusion :</span>**
<br>  on a cinq <strong>1</strong> négative, peut être parceque notre data set ils ne sont pas balancé est notre score il ne sont pas $ 100$%
  <center>-----------------------------------------</center>

In [19]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[   0    5]
 [   0 1472]]
              precision    recall  f1-score   support

           0       0.00      0.00      0.00         5
           1       1.00      1.00      1.00      1472

   micro avg       1.00      1.00      1.00      1477
   macro avg       0.50      0.50      0.50      1477
weighted avg       0.99      1.00      0.99      1477

0.996614759647935


C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


In [20]:
knn = KNeighborsClassifier(n_neighbors=2)
knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[   0    5]
 [   3 1469]]
              precision    recall  f1-score   support

           0       0.00      0.00      0.00         5
           1       1.00      1.00      1.00      1472

   micro avg       0.99      0.99      0.99      1477
   macro avg       0.50      0.50      0.50      1477
weighted avg       0.99      0.99      0.99      1477

0.994583615436696


In [21]:
from sklearn.tree import DecisionTreeClassifier
classifier = DecisionTreeClassifier()
classifier.fit(X_train, y_train)
y_pred=classifier.predict(X_test)
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[   5    0]
 [   0 1472]]
              precision    recall  f1-score   support

           0       1.00      1.00      1.00         5
           1       1.00      1.00      1.00      1472

   micro avg       1.00      1.00      1.00      1477
   macro avg       1.00      1.00      1.00      1477
weighted avg       1.00      1.00      1.00      1477

1.0


>* **<span style="color:#0000C0">Conclusion:</span>**
><ol><li>j'ai conclur en utilisant la précision comme mesure de performance, le modèle que le modèle Decision Tree Classifier avec deux classe est plus précis 
 $(1.0)$ que le modèle KNeighborsClassifier $(0,99)$ et Logistic Regression $(0,99)$</li>
    <li>plus que le data set est équilibré plus quelle la précision de l'algorithme est forte </li></ol>

> <strong>Dans cette partie je veux modifer les données de telle façon les donnés devient balancé 😎</strong>

* Tout d'abors j'ai changé la colonne df2['target'] de telles façons les nombres de $un$ et les $deux$ sont égaux

In [22]:
def rand_bin_array(K, N):
    arr = np.zeros(N)
    arr[:K]  = 1
    np.random.shuffle(arr)
    return arr

In [23]:
df2.count()

Numero de mesure           4922
Longitude                  4922
Latitude                   4922
Position                   4922
Insee                      4922
Adresse                    4922
Complement                 4404
Code postal                4922
Commune                    4922
Date de mesure             4922
Mesure realisee            4922
Rapport                    4922
Conformite de la mesure    4922
Niveau_global              4922
Type d'environnement       4922
Laboratoire                4922
target                     4922
dtype: int64

In [24]:
arr=rand_bin_array(2461, 4922)
print(arr)
target=df2['target'].to_numpy()
df2['target']=arr
print(df2['target'])

[0. 0. 0. ... 1. 0. 0.]
0       0.0
1       0.0
2       0.0
3       1.0
4       0.0
       ... 
4917    0.0
4918    0.0
4919    1.0
4920    0.0
4921    0.0
Name: target, Length: 4922, dtype: float64


In [25]:
pd.crosstab(df2.target, "freq")

col_0   freq
target      
0.0     2461
1.0     2461

* **<span style="color:#0000C0">Maintemnet j'ai continué avec le même etape
 :</span>**

In [26]:
X=df2[['Numero de mesure','Niveau_global']].values
print(X)
y=df2['target'].to_numpy()
print(y)

[[1.75000000e+02 0.00000000e+00]
 [1.80000000e+02 0.00000000e+00]
 [5.43000000e+02 5.40000000e-01]
 ...
 [3.58600000e+03 4.74652768e+00]
 [2.95600000e+03 4.09413697e+00]
 [2.72800000e+03 1.70854568e+00]]
[0. 0. 0. ... 1. 0. 0.]


In [27]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=0)

In [28]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[351 385]
 [372 369]]
              precision    recall  f1-score   support

         0.0       0.49      0.48      0.48       736
         1.0       0.49      0.50      0.49       741

   micro avg       0.49      0.49      0.49      1477
   macro avg       0.49      0.49      0.49      1477
weighted avg       0.49      0.49      0.49      1477

0.48747461069735953


In [29]:
knn = KNeighborsClassifier(n_neighbors=2)
knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[531 205]
 [558 183]]
              precision    recall  f1-score   support

         0.0       0.49      0.72      0.58       736
         1.0       0.47      0.25      0.32       741

   micro avg       0.48      0.48      0.48      1477
   macro avg       0.48      0.48      0.45      1477
weighted avg       0.48      0.48      0.45      1477

0.4834123222748815


In [30]:
from sklearn.tree import DecisionTreeClassifier
classifier = DecisionTreeClassifier()
classifier.fit(X_train, y_train)
y_pred=classifier.predict(X_test)
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[388 348]
 [355 386]]
              precision    recall  f1-score   support

         0.0       0.52      0.53      0.52       736
         1.0       0.53      0.52      0.52       741

   micro avg       0.52      0.52      0.52      1477
   macro avg       0.52      0.52      0.52      1477
weighted avg       0.52      0.52      0.52      1477

0.5240352064996615


In [31]:
from sklearn.linear_model import LogisticRegression 
#Create a l.Reg Classifier
logreg1 = LogisticRegression()
#Train the model using the training sets
logreg1.fit(X_train, y_train)
y_pred=logreg1.predict(X_test)
print(confusion_matrix(y_test,y_pred))
print(classification_report(y_test,y_pred))
print(accuracy_score(y_test, y_pred))

[[  0 736]
 [  0 741]]
              precision    recall  f1-score   support

         0.0       0.00      0.00      0.00       736
         1.0       0.50      1.00      0.67       741

   micro avg       0.50      0.50      0.50      1477
   macro avg       0.25      0.50      0.33      1477
weighted avg       0.25      0.50      0.34      1477

0.5016926201760324


C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:433: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
C:\Users\ASUS\Anaconda3\lib\site-packages\sklearn\metrics\classification.py:1143: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)


<strong>Malheureusement ma réflexion est fausse, lorsqu ont équilibré les donnés l'algorithme ML subi des difficultés de faire de prédiction pour cela le score est 50% </strong>


>* **<span style="color:#0000C0">Conclusion:</span>**
><ol><li>j'ai conclur en utilisant la précision comme mesure de performance, le modèle $KNN$ avec deux classe est plus précis 
 $(0,51)$ que le modèle Decision Tree Classifier $(0,50)$ et Logistic Regression $(0,40)$</li>
    <li>plus que le data set est équilibré plus quelle la précision de l'algorithme est faible </li></ol>

In [1]:
import pandas as pd 
import numpy as np
import csv
import matplotlib.pyplot as plt
import seaborn as sns
from csv import writer
import random
import sklearn
%load_ext watermark


sklearn 0.20.1
csv     1.0
pandas  1.1.3
numpy   1.17.4
seaborn 0.9.0

